wp/lib/bap_wp/src/constraint.ml

```diff
@@ -312,9 +366,9 @@ let get_refuted_goals ?filter_out:(filter_out = []) (constr : t)
       let hyps_false =
         let hyp_vals =
           List.map hyps ~f:(fun h -> eval_model_exn model (eval_aux h olds news ctx)) in
-          if (List.for_all hyp_vals ~f:(fun e -> Z3.Boolean.is_true e || Z3.Boolean.is_false e))
-          then not (List.for_all hyp_vals ~f:Z3.Boolean.is_true)
-          else failwith "Unexpected Expression processing Clause"
+        if (List.for_all hyp_vals ~f:(fun e -> Z3.Boolean.is_true e || Z3.Boolean.is_false e))
+        then not (List.for_all hyp_vals ~f:Z3.Boolean.is_true)
```

Comment:
Can we use a `List.exists ...` instead of a `not (List.forall ...)`? I think it might be a bit clearer.